crates/sui-config/src/genesis.rs

```diff
@@ -58,6 +59,16 @@ pub struct Genesis {
     validator_set: Vec<ValidatorInfo>,
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct GenesisTuple(
+    pub ProtocolVersion,
```

Comment:
do we need this? the protocol version should be encoded in the system object so we potentially could read it from there (it would take slightly more work of course)

Reply:
Its basically here as an error check. When the intermediate genesis data is loaded (e.g. when adding validators, verifying) we want to make sure that the serialized version is that same as whatever the user has selected / defaulted to.

Reply:
It seems redundant because if the version is in the system object, and the same version wasn't used, then the comparison that exists would already fail

Reply:
ok, yeah, we may be able to remove it. i'll look into it.

Reply:
fixed. i left in a check against mismatches, which is technically unnecessary (you could in theory sign a blob with a protocol version that isn't even supported by your `sui` binary, but that most likely indicates that you have built `sui` at the wrong commit).